base-template.sh

```diff
@@ -592,6 +590,60 @@ call_script() {
     return $?
 }
 
+#####################################################
+# Show a prompt with the text `$2` with
+#   hint text `$3` with
+#   options `$4` in form of e.g. `Y/a/n/d` and
+#   optional long options [optional]:
+#   e.g.
+#    `$5-$8` : "Yes" "All" "None" "Disable"
+#   First capital short option character is treated
+#   as default.
+#
+#####################################################
+show_prompt() {
+    DIALOG_TOOL="$(get_tool_script "dialog")"
+    VARIABLE="$1"
+    shift
+    TEXT="$1"
+    shift
+    HINT_TEXT="$1"
+    shift
+    SHORT_OPTIONS="$1"
+    shift
+
+    if [ "$DIALOG_TOOL" != "" ]; then
+        ANSWER=$(call_script "$DIALOG_TOOL" "githook::" "$TEXT" "$SHORT_OPTIONS" "$@")
+        # shellcheck disable=SC2181
+        if [ $? -eq 0 ]; then
+
+            if ! echo "$SHORT_OPTIONS" | grep -q "$ANSWER"; then
+                echo "! Dialog tool did return wrong answer $ANSWER -> Abort."
+                exit 1
+            fi
+
+            # Safeguard `eval`
+            if ! echo "$VARIABLE" | grep -qE "^[A-Z_]+\$"; then
+                echo "! Invalid variable name: $VARIABLE"
+                exit 1
+            fi
+
+            eval "$VARIABLE"="$ANSWER"
+            return
+        elif [ $? -ge 2 ]; then
```

Comment:
Can we just have all failures fall back to terminal please?

Reply:
we can  do that. yes.